Changes to notebooks 4-8 and 11

peterverhaar opened the pull request:
In notebooks 4-8, the exercises have been moved to the sections in the tutorial explaining the topics needed to do the exercises. A number of new exercises have also been added. I also made a minor textual revision in notebook 11 (the instruction in exercise 11.1. to give the dataframe a name that is distinct from that of the dataframe used in the tutorial)

peterverhaar committed:
Changes to notebooks 4-8 and 11

bencomp commented on notebooks/8 Working_with_files_and_folders.ipynb:
```diff
@@ -160,60 +170,57 @@
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "import os\n",
+    "\n",
+    "dir = 'Solutions'\n",
+    "\n",
+    "for fileName in os.listdir( dir ):\n",
+    "    print( fileName )"
+   ]
```
`dir` is a built-in Python function.
```suggestion
   "source": [
    "import os\n",
    "\n",
    "directory = 'Solutions'\n",
    "\n",
    "for file_name in os.listdir( directory ):\n",
    "    print( file_name )"
   ]
```

bencomp commented on notebooks/8 Working_with_files_and_folders.ipynb:
```diff
@@ -160,60 +170,57 @@
-    "Download the following file:\n",
-    "\n",
-    "[https://edu.nl/ggwtf](https://edu.nl/ggwtf)\n",
+    "The function `join()`, which is part of the `path` module of `os`, can be used to create a string representing the path to a certain file. If you have one variable which records the base directory of a file, and a second variable which captures the filename, the `join()` function can concatenate the values of these two variables to create the full path to this file. The `join()` function alw ...
```
```suggestion
    "The function `join()`, which is part of the `path` module of `os`, can be used to create a string representing the path to a certain file. If you have one variable which records the base directory of a file, and a second variable which captures the filename, the `join()` function can concatenate the values of these two variables to create the full path to this file. The `join()` function always follows the conventions that are in place on a given operating system for representing paths. There can often be certain differences. While Mac OS uses forward slashes, for instance, Windows uses back slashes. Working with `join()` makes your code more platform-independent.\n",
```

bencomp commented:
Nice! Working on the review!

bencomp commented on notebooks/8 Working_with_files_and_folders.ipynb:
````diff
@@ -160,60 +170,57 @@
-    "1. [line1]\n",
-    "2. [line2]\n",
-    "```\n"
+    "The code below offers a demostration of these two functions. It lists all the files in the directory that is mentiond, and makes sure that all the subdirectories are ignored. Note that the first line imports the two functions that have been discussed above. As a result of this, it is no longer necessary to use the period syntax for `isfile()` and `join()`.  "
````
```suggestion
    "The code below offers a demonstration of these two functions. It lists all the files in the directory that is mentioned, and makes sure that all the subdirectories are ignored. Note that the first line imports the two functions that have been discussed above. As a result of this, it is no longer necessary to use the period syntax for `isfile()` and `join()`."
```

bencomp commented on notebooks/8 Working_with_files_and_folders.ipynb:
```diff
@@ -222,32 +229,26 @@
+    "from os.path import isfile , join\n",
     "\n",
-    "response = requests.get('https://edu.nl/ggwtf')\n",
+    "dir = 'Solutions'\n",
```
```suggestion
    "directory = 'Solutions'\n",
```

bencomp commented on notebooks/8 Working_with_files_and_folders.ipynb:
```diff
@@ -222,32 +229,26 @@
-    "    contents = response.text\n",
-    "    with open('sonnet116.txt' , 'w' , encoding = 'utf-8') as fh:\n",
-    "        fh.write(contents)\n"
+    "for fileName in os.listdir( dir ):\n",
+    "    if isfile( join( dir , fileName ) ):\n",
+    "        print( fileName ) \n"
```
```suggestion
    "for file_name in os.listdir( directory ):\n",
    "    if isfile( join( directory , file_name ) ):\n",
    "        print( file_name ) \n"
```

bencomp commented on notebooks/Solutions/4 Flow_control.ipynb:
```diff
@@ -142,19 +179,15 @@
+    "The `randint()` method from `random` can be used, firstly, to generate a random integer. \n",
     "\n",
-    "When the user enters a value which is too low or too high, this information is communicated to the user via a print statement. To test whether the number that is guesses is correct, you can work with the equality (`==`) and the inequality (`!=`) operators.\n",
+    "Use the function `input()` to request a value from the user. The function int() needs to be used to convert the input into an integer.\n",
```
```suggestion
    "Use the function `input()` to request a value from the user. The function `int()` needs to be used to convert the input into an integer.\n",
```

bencomp commented on notebooks/Solutions/4 Flow_control.ipynb:
```diff
@@ -142,19 +179,15 @@
-    "Hint: you will need to combine the different kinds of flow control to complete the whole game.\n",
-    "Start by checking if the guessed number is correct and making sure that this part of the code works as you expect.\n",
-    "When that works, think which parts of the code need to be repeated to let the player guess until the number was found."
+    "When the user enters a value which is too low or too high, this information is communicated to the user via a print statement. To test whether the number that is guesses is correct, you can work with the equality (`==`) and the inequality (`!=`) operators. "
```
```suggestion
    "When the user enters a value which is too low or too high, this information is communicated to the user via a print statement. To test whether the number that is guesses is correct, you can work with the equality (`==`) and the inequality (`!=`) operators.\n",
    "\n",
    "Hint: you will need to combine the different kinds of flow control to complete the whole game.\n",
    "Start by checking if the guessed number is correct and making sure that this part of the code works as you expect.\n",
    "When that works, think which parts of the code need to be repeated to let the player guess until the number was found."
```

bencomp commented on notebooks/Solutions/7 Functions_modules_and_libraries.ipynb:
```diff
@@ -11,7 +11,32 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def celciusToFahrentheit( celcius ):\n",
```
```suggestion
    "def celcius_to_fahrentheit( celcius ):\n",
```

bencomp commented on notebooks/Solutions/7 Functions_modules_and_libraries.ipynb:
```diff
@@ -11,7 +11,32 @@
+    "    fahrenheit = 1.8 * celcius + 32 \n",
+    "    return round( fahrenheit , 1 )\n",
+    "\n",
+    "print( '{} degrees Celcius is {} degrees in Fahrenheit'.format( 20 , celciusToFahrentheit(20) ))\n",
+    "print( '{} degrees Celcius is {} degrees in Fahrenheit'.format( 37 , celciusToFahrentheit(37)  ))"
```
```suggestion
    "print( '{} degrees Celcius is {} degrees in Fahrenheit'.format( 20 , celcius_to_fahrentheit(20) ))\n",
    "print( '{} degrees Celcius is {} degrees in Fahrenheit'.format( 37 , celcius_to_fahrentheit(37)  ))"
```

bencomp commented on notebooks/Solutions/7 Functions_modules_and_libraries.ipynb:
```diff
@@ -79,24 +95,40 @@
-    "from math import *\n",
+    "numbers = [6,67,8,33,24,11,34,6,23,4,19,6,9,12,134,45,23]\n",
     "\n",
+    "def calculate_average(numbers,nr_digits=2):\n",
+    "    average = sum(numbers)/len(numbers)\n",
+    "    return round(average,nr_digits)\n",
```
"The default value must be 1." 😄 
```suggestion
    "def calculate_average(numbers, nr_digits=1):\n",
    "    average = sum(numbers)/len(numbers)\n",
    "    return round(average, nr_digits)\n",
```

bencomp commented on notebooks/Solutions/8 Working_with_files_and_folders.ipynb:
````diff
@@ -75,41 +57,41 @@
-    "```"
+    "Find a directory on your own computer which contains both files and subfolders. Define the path to this folder, as a variable named `path`.\n",
+    "\n",
+    "Next, write some code that can list only the subdirectories in this directories, and which ignores the individual files, using the functions `listdir()` and `isFile`. \n",
````
```suggestion
    "Next, write some code that can list only the subdirectories in this directories, and which ignores the individual files, using the functions `listdir()` and `isfile`. \n",
```

bencomp reviewed (approved):
I will commit my suggestions and merge.

bencomp committed:
Apply suggestions from code review

Mostly minor edits to normalise or fix spacing, formatting, Pythonic naming, and typos.
